tensorflow_addons/optimizers/constants.py

```diff
@@ -0,0 +1,5 @@
+import tensorflow as tf
+
+BASE_OPTIMIZER_CLASS = tf.keras.optimizers.legacy.Optimizer
```

Comment:
Why not if ... else? It will crash if tf.keras.optimizers.legacy doesn't exist (TF 2.8), right?

Are you sure it has to be CAPITAL_LETTER?

I'd rename it to KerasLegacyOptimizer (or KERAS_LEGACY_OPTIMIZER - I'm not an expert in conventions :smile: ) and rename the file to keras.py.

Reply:
Thanks! I changed the check condition.

For the naming, yea, we probably want to do camel case since it is a class itself. But I don't want to imply Legacy in the name, which is not yet 100% correct - we still use `tf.keras.optimizers.Optimizer` in many places. I am renaming to `BaseOptimizerClass`, wdyt?

Reply:
@chenmoneygithub - I expect that after refactor in keras is done, we will be gradually switching back to `tf.keras.optimizers.Optimizer`. `BaseOptimizerClass` will be confusing then because it'll be name for the legacy class.

Reply:
sg, renamed to KerasLegacyOptimizer.